src/PointwiseFunctions/AnalyticData/Xcts/BinaryWithGravitationalWaves.cpp

```diff
@@ -0,0 +1,756 @@
+    const gsl::not_null<Scalar<DataType>*> distance_left,
+    const gsl::not_null<Cache*> /*cache*/,
+    detail::Tags::DistanceLeft<DataType> /*meta*/) const {
+  tnsr::I<DataType, 3> v(x);
+  v.get(0) -= xcoord_left;
+  *distance_left = magnitude(v);
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    const gsl::not_null<Scalar<DataType>*> distance_right,
+    const gsl::not_null<Cache*> /*cache*/,
+    detail::Tags::DistanceRight<DataType> /*meta*/) const {
+  tnsr::I<DataType, 3> v(x);
+  v.get(0) -= xcoord_right;
+  *distance_right = magnitude(v);
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    const gsl::not_null<tnsr::i<DataType, 3>*> deriv_one_over_distance_left,
+    const gsl::not_null<Cache*> cache,
+    ::Tags::deriv<detail::Tags::OneOverDistanceLeft<DataType>,
+                  tmpl::size_t<Dim>, Frame::Inertial> /*meta*/) const {
+  const auto& distance_left =
+      get(cache->get_var(*this, detail::Tags::DistanceLeft<DataType>{}));
+  const auto& normal_left =
+      cache->get_var(*this, detail::Tags::NormalLeft<DataType>{});
+  for (size_t i = 0; i < 3; ++i) {
+    deriv_one_over_distance_left->get(i) =
+        -normal_left.get(i) / square(distance_left);
+  }
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    const gsl::not_null<tnsr::i<DataType, 3>*> deriv_one_over_distance_right,
+    const gsl::not_null<Cache*> cache,
+    ::Tags::deriv<detail::Tags::OneOverDistanceRight<DataType>,
+                  tmpl::size_t<Dim>, Frame::Inertial> /*meta*/) const {
+  const auto& distance_right =
+      get(cache->get_var(*this, detail::Tags::DistanceRight<DataType>{}));
+  const auto& normal_right =
+      cache->get_var(*this, detail::Tags::NormalRight<DataType>{});
+  for (size_t i = 0; i < 3; ++i) {
+    deriv_one_over_distance_right->get(i) =
+        -normal_right.get(i) / square(distance_right);
+  }
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    gsl::not_null<tnsr::ijk<DataType, 3>*> deriv_3_distance_left,
+    gsl::not_null<Cache*> cache,
+    ::Tags::deriv<
+        ::Tags::deriv<::Tags::deriv<detail::Tags::DistanceLeft<DataType>,
+                                    tmpl::size_t<Dim>, Frame::Inertial>,
+                      tmpl::size_t<Dim>, Frame::Inertial>,
+        tmpl::size_t<Dim>, Frame::Inertial> /*meta*/) const {
+  const auto& distance_left =
+      get(cache->get_var(*this, detail::Tags::DistanceLeft<DataType>{}));
+  const auto& normal_left =
+      cache->get_var(*this, detail::Tags::NormalLeft<DataType>{});
+  std::array<std::array<double, 3>, 3> delta{
+      {{{1., 0., 0.}}, {{0., 1., 0.}}, {{0., 0., 1.}}}};
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      for (size_t k = 0; k < 3; ++k) {
+        deriv_3_distance_left->get(i, j, k) =
```

Comment:
(optional) You could make some of these expressions easier to implement using our tensor expressions framework. See `src/Evolution/Systems/CurvedScalarWave/Worldtube/KerrSchildDerivatives.cpp` for examples. I don't think a Kronecker delta is supported yet though, right @macedo22?

Reply:
Correct, Kronecker delta is not yet supported